linode/ipv6range/framework_models.go

```diff
@@ -0,0 +1,69 @@
+package ipv6range
+
+import (
+	"context"
+	"encoding/json"
+
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/linode/linodego"
+)
+
+type DataSourceModel struct {
+	Range   types.String `tfsdk:"range"`
+	IsBGP   types.Bool   `tfsdk:"is_bgp"`
+	Linodes types.Set    `tfsdk:"linodes"`
+	Prefix  types.Int64  `tfsdk:"prefix"`
+	Region  types.String `tfsdk:"region"`
+	ID      types.String `tfsdk:"id"`
+}
+
+type ResourceModel struct {
+	PrefixLength types.Int64  `tfsdk:"prefix_length"`
+	LinodeId     types.Int64  `tfsdk:"linode_id"`
+	RouteTarget  types.String `tfsdk:"route_target"`
+	Range        types.String `tfsdk:"range"`
+	IsBGP        types.Bool   `tfsdk:"is_bgp"`
+	Linodes      types.Set    `tfsdk:"linodes"`
+	Region       types.String `tfsdk:"region"`
+	ID           types.String `tfsdk:"id"`
+}
+
+func (data *DataSourceModel) parseIPv6Range(
```

Comment:
I think we'll need to apply the same split-logic parse functions as discussed here: https://github.com/linode/terraform-provider-linode/pull/911#discussion_r1251153929